discovery/src/main/java/com/ibm/watson/developer_cloud/discovery/v1/model/QueryAggregation.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2017 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.ibm.watson.developer_cloud.discovery.v1.model;
+
+import java.util.List;
+
+import com.google.gson.annotations.SerializedName;
+import com.ibm.watson.developer_cloud.service.model.GenericModel;
+
+/**
+ * An aggregation produced by the Discovery service to analyze the input provided.
+ */
+public class QueryAggregation extends GenericModel {
+
+  /** The type of aggregation command used. e.g. term, filter, max, min, etc. */
+  private String type;
+  /** The field where the aggregation is located in the document. */
+  private String field;
+  private List<AggregationResult> results;
+  /** The match the aggregated results queried for. */
+  private String match;
+  /** Number of matching results. */
+  @SerializedName("matching_results")
+  private Long matchingResults;
+  /** Aggregations returned by the Discovery service. */
+  private List<QueryAggregation> aggregations;
+  /** Interval specified by using aggregation type 'timeslice'. */
+  private Long interval;
```

Comment:
when it is a timeslice, `interval` is a `String`, not a `Long` -> when it is a histogram, it is a `Long` - best solution is to make this an `Object` since we aren't going to support polymorphic models